sympy/polys/polyutils.py

```diff
@@ -232,6 +232,8 @@ def _parallel_dict_from_expr_if_gens(exprs, opt):
                     except KeyError:
                         if not factor.has_free(*opt.gens):
                             coeff.append(factor)
+                        elif opt.extension is True and factor.is_algebraic:
+                            coeff.append(factor)
```

Comment:
This is a bug in `has_free`:
```python
In [1]: RootOf(x**3 + x + 1, 0).has_free(x)
Out[1]: True
```
It should be fixed there rather than worked around here.

Reply:
cf #26637 and #26808